### DISCUSSION TITLE
functions with (interface{}, error) in env are failing

### DISCUSSION THREAD

**Discussion by neilmi** [antonmedv/expr#329] [Q&A]
seems like we can't update and use function in env as we used to...
env only gets interface{} return type but it seems that the latest version doesn't have expr.Function yet..

when can we expect this to be released?

**Comment by antonmedv** [comment]
Will take a look today evening.

**Comment by antonmedv** [comment]
Please share failing example.

**Reply by neilmi** [reply]
hey anton thanks for the quick reply
that's what i get:
![Screen Shot 2023-02-01 at 17 28 11](https://user-images.githubusercontent.com/102041588/216086408-41f2a839-a2a2-45c4-988d-886b820089fc.png)

that's my function signature
![Screen Shot 2023-02-01 at 17 28 33](https://user-images.githubusercontent.com/102041588/216086521-56fd00ae-d287-493b-95fd-93f796107bb6.png)


when i remove error from return types, its ok..

**Reply by antonmedv** [reply]
Will take a look. 

BTW, what if we add if-then-else operator to expr?

**Reply by neilmi** [reply]
what do you mean? the problem happens with our functions..
unfortunately we are heavily relay on if function in a lot of places haha, so we better stick with this...

thanks a lot, really appreciate this

**Comment by antonmedv** [comment]
Fixed in v1.11

**Reply by neilmi** [reply]
wow that was fast! 
appreciate that!! thx
